lib/get.joi.js

```diff
@@ -0,0 +1,21 @@
+function getJoi() {
```

Comment:
I am thinking we should use more idomatic ES6:
```javascript
const softRequire = (path) => {
  try {
    return require(path);
  } catch (ignoreError) {
    return undefined;
  }
}

const getJoi = () => {
  const joi = softRequire('joi') || softRequire('@hapi/joi');

  if (!joi) {
    throw new Error('joi-password-complexity requires either `joi` or `@hapi/joi` to be installed.');
  }

  return joi;
}

module.exports = getJoi;
```

Reply:
I agree, tried this one but eslint complained due to no-dynamic-import, saying that `require(...)` must take a string literal, not a parameter. Would you rather relax the eslint rule?

Also, as to the arrow function/regular function, I personally prefer function declarations because they allow you to write code [step-down style](https://dzone.com/articles/the-stepdown-rule), but it's your call really.

Reply:
Yeah let's relax the rule. I can take care of that after the merging.
As for step-down, while I have no issue with the style in general, in some languages you can't actually do that.

C# you can do it because all functions at the same level (class scope, or classes in a namespace) have already been declared before the code inside of them is parsed. Almost as if there is an inherent header file.
This is the same for `function` declarations in javascript.

Unfortunately in other languages, you simply can't do step-down. For this reason, I don't find it convincing enough to _enforce_. I think it makes code readable, but you will get into organization knots trying to do this. I have this problem at work with Java/Kotlin right now, where we end up "pulling out" abstractions to a higher level so it is more readable and then you end up with hundreds (sometimes thousands) of classes, sometimes with methods that simply call another method with a variant of parameters in order to enforce readable method naming and step-down abstraction layering. It actually _looks good_, until you need to refactor... then it becomes a disgusting nightmare (at least with Java/Kotlin).

For now, since this is so simple, I am going to say let's skip on the step-down. if there becomes an issue where we need that specific paradigm for readability, I'll be happy to change it.

Reply:
Well, glad JS supports this then.

Thanks for the elaborate response, really appreciate it.